src/jdk.compiler/share/classes/com/sun/tools/javac/util/Log.java

```diff
@@ -261,12 +262,27 @@ protected Log(Context context) {
      * @param context the context in which to find writers to use
      * @return a map of writers
      */
+    private final static Charset nativeCharset;
+    static {
+        Charset cs = Charset.defaultCharset();
```

Comment:
This could move into the `catch` section as a last resort.

Reply:
Move `cs = Charset.defaultCharset()` into `catch` section